src/index.ts

```diff
@@ -0,0 +1,58 @@
+import * as  amqp from 'amqplib/callback_api';
+import { Channel, Connection, Message } from 'amqplib/callback_api';
+import { config } from 'dotenv'; config(); //initialize right there the configuration
+import { readFileSync } from 'fs';
+import { QueueMessage } from './types';
+import { PayloadSchema } from './schemas';
+import { connectToDatabase } from './datasource';
+import { Logger } from './utils';
+import * as Joi from 'joi';
+const utils = require('utils')._;
+import {
+  recordStartMigration,
+  getClientId,
+  recordStructureValidationStatus,
+}
+  from './modules';
+
+amqp.connect(process.env.AVW_QUEUE_HOST || 'amqp://localhost', function (error: Error, connection: Connection) {
+  connection.createChannel(function (error: Error, channel: Channel) {
+    var queueName = process.env.AVW_QUEUE_NAME || "DHIS2_VALIDATION_QUEUE";
+    channel.assertQueue(queueName, { durable: true });
+    console.log("[*] Waiting for messages on %s. To exit press CTRL+C", queueName);
+    channel.consume(queueName, async function (message: Message | null) {
+      if (message) {
```

Comment:
start with the negative null case and end with the happy path (again, no need for an else statement)